scripts/binary-validation.sh

```diff
@@ -0,0 +1,15 @@
+#!/bin/sh
+
+./gradlew apiCheck -q
+EXIT_CODE=$?
+if [ $EXIT_CODE -ne 0 ]; then
+  echo "❌ apiCheck failed, running apiDump for you..."
+
+  ./gradlew apiDump -q
+
+  echo "API dump done, please check the results and then try your commit again!"
```

Comment:
Along with my comment above, I think it would be good to explain more of what "check the results" means. 

```suggestion
  echo "API dump done, please check the modified '.api' files to make sure that the 'public' code changes that were done were done on purpose and not by accident. If done on purpose, please try committing again!"
```